src/main/java/updater/UpdateManager.java

```diff
@@ -0,0 +1,143 @@
+package updater;
+
+import javafx.application.Platform;
+import javafx.scene.control.Menu;
+import javafx.scene.control.MenuItem;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import ui.TestController;
+import ui.UI;
+import util.DialogMessage;
+import util.events.UpdateManagerProgressWindowEventHandler;
+
+import java.io.File;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.net.URL;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+
+/**
+ * UpdateManager is the class that will handle updating of HubTurbo application
+ */
+public class UpdateManager {
+    private static final Logger logger = LogManager.getLogger(UpdateManager.class.getName());
+
+    private static final String UPDATE_DIRECTORY = "updates";
+
+    private static final String UPDATE_SERVER_DATA_NAME =
+            "https://raw.githubusercontent.com/HubTurbo/AutoUpdater/master/HubTurbo.xml";
+    private static final String UPDATE_LOCAL_DATA_NAME = "HubTurbo.json";
+
+    private static final String UPDATE_APP_NAME = "HubTurbo.jar";
+
+    private FileDownloader fileDownloader;
+
+    private final ExecutorService pool = Executors.newSingleThreadExecutor();
+
+    public UpdateManager() {
+        fileDownloader = new FileDownloader();
+    }
+
+    /**
+     * Driver method to trigger UpdateManager to run. Update will be run on another thread.
+     *
+     * - Run is not automatic upon instancing the class in case there would like to be conditions on when to run update,
+     *   e.g. only if user is logged in
+     */
+    public void run() {
+        if (!TestController.isTestMode()) {
```

Comment:
@dariusf I added this here so that update manager won't run in test mode (since each test will try to download update and it may send repeated requests to GitHub server). Is this okay or do you have other suggestion in mind?